libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/FileAccessTree.java

```diff
@@ -75,8 +102,8 @@ static String normalizePath(Path path) {
         return path.toAbsolutePath().normalize().toString();
     }
 
-    private static boolean checkPath(String path, String[] paths) {
-        if (paths.length == 0) {
+    private boolean checkPath(String path, String[] paths) {
+        if (paths.length == 0 || Arrays.binarySearch(exclusivePaths, path) >= 0) {
```

Comment:
Can someone configure exclusive access to a directory? If so, this check is insufficient I think.

Reply:
Please double check this logic for me -- I intend to throw in the constructor for `FileAccessTree` if we catch a read or read_write path that would require exclusive access to a directory for a read or read_write path. In which case, that leaves the exact match for an exclusive path in an existing read or read_write directory or it will already have thrown in the constructor.

Reply:
Gosh. Tricky.

Ok, suppose:
- X has read access to `/a`
- Y has exclusive access to `/a/b`
- X attempts to read `/a/b/c`

I believe this combo would not be caught in the constructor (because it's valid) and we would fail to detect the violation because it's not an exact file name match.

Reply:
Ah, I will add this test and move the logic :)

Reply:
Fixed this. Thanks for showing me what was incorrect with a simple example.